src/java/edu/slu/action/ObjectAction.java

```diff
@@ -811,28 +843,222 @@ public void updateObject() throws IOException, ServletException, Exception{
+         boolean detectedPrevious = !previous_id.equals("");
+         //Update the history.previous of all the next ids in the array of the deleted object
+         for(int n=0; n<next_ids.size(); n++){
+             BasicDBObject query = new BasicDBObject();
+             BasicDBObject objToUpdate;
+             BasicDBObject objWithUpdate;
+             String nextID = next_ids.getString(n);
+             query.append("@id", nextID);
+             objToUpdate = (BasicDBObject) mongoDBService.findOneByExample(Constant.COLLECTION_ANNOTATION, query); 
+             if(null != objToUpdate){
+                JSONObject fixHistory = JSONObject.fromObject(objToUpdate);
+                if(isRoot){ //The object being deleted was root.  That means these next objects must become root.  Strictly, all history trees must have num(root) > 0.  
+                    fixHistory.getJSONObject("__rerum").getJSONObject("history").element("prime", "root");
+                    newTreePrime(fixHistory);
+                    //Do not overwrite previous of the new prime objects so they reference the deleted prime object for the future.
+                    //fixHistory.getJSONObject("__rerum").getJSONObject("history").element("previous", "");
+                }
+                else if(detectedPrevious){ //The object being deleted had a previous.  That is now absorbed by this next object to mend the gap.  
+                    fixHistory.getJSONObject("__rerum").getJSONObject("history").element("previous", previous_id);
+                }
+                else{
+                    //Yikes this is some kind of error...it is either root or has a previous, this case means neither are true.
+                    success = false;
+                }
+                Object forMongo = JSON.parse(fixHistory.toString()); //JSONObject cannot be converted to BasicDBObject
+                objWithUpdate = (BasicDBObject)forMongo;
+                mongoDBService.update(Constant.COLLECTION_ANNOTATION, objToUpdate, objWithUpdate);
+             }
+             else{
+                 success = false;
+                 //Yikes this is an error, could not find an object assosiated with id found in history tree.
+             }
+         }
+         if(detectedPrevious){ 
+             //The object being deleted had a previous.  That previous object next[] must be updated with the deleted object's next[].
+             BasicDBObject query2 = new BasicDBObject();
+             BasicDBObject objToUpdate2;
+             BasicDBObject objWithUpdate2;
+             query2.append("@id", previous_id);
+             objToUpdate2 = (BasicDBObject) mongoDBService.findOneByExample(Constant.COLLECTION_ANNOTATION, query2); 
+             if(null != objToUpdate2){
+                JSONObject fixHistory2 = JSONObject.fromObject(objToUpdate2);  
+                fixHistory2.getJSONObject("__rerum").getJSONObject("history").element("next", next_ids);
```

Comment:
What needs to happen is:

fixHistory2.__rerum.history.next needs to have the DO.@id sliced out and the DO.__rerum.history.next array concatenated.